op-program/client/l2/engine.go

```diff
@@ -94,7 +98,7 @@ func (o *OracleEngine) PayloadByHash(ctx context.Context, hash common.Hash) (*et
 	if block == nil {
 		return nil, ErrNotFound
 	}
-	return eth.BlockAsPayloadEnv(block, o.rollupCfg.CanyonTime)
+	return eth.BlockAsPayloadEnv(block, o.backend.Config().ShanghaiTime)
```

Comment:
Based on the spec, Canyon (not `Cancun`) already contains Shanghai features - https://specs.optimism.io/protocol/canyon/overview.html. I might be missing something, but this change doesn't seem necessary to me.

Reply:
One important detail here as well is that the way it works (even in the spec) isn't strictly that Canyon activates these features but that ShanghaiTime must be set to the same as CanyonTime.